### PR TITLE
Adds complete ProtBERT

### PR DESCRIPTION
**Pull request by Shiva-sankaran**
## Description

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by NimishaDey** [comment]
CI needs to be checked. I see some errors related to test case.

File: deepchem/models/torch_models/prot_bert.py [hunk 1]
```diff
@@ -41,12 +41,13 @@ class ProtBERT(HuggingFaceModel):
     >>> from deepchem.models.torch_models.prot_bert import ProtBERT
     >>> pretrain_model_dir = os.path.join(tempdir, 'pretrain-model')
     >>> model_path = 'Rostlab/prot_bert'
-    >>> pretrain_model = ProtBERT(task='mlm', HG_model_path=model_path, n_tasks=1, model_dir=pretrain_model_dir)  # mlm pretraining
+    >>> pretrain_model = ProtBERT(task='mlm', model_pretrain_dataset='uniref100', n_tasks=1, model_dir=pretrain_model_dir)  # mlm pretraining
```
Comment (rbharath) [inline deepchem/models/torch_models/prot_bert.py:44]: I think this change makes it more rigid and less flexible. Let's revert to the previous design and add a comment explaining possible models

**Review by rbharath — COMMENTED** [review]
Can we add the sequence classification datasets to the deepchem S3 and repro with this model?

**Comment by Shiva-sankaran** [comment]
Closing as #4052 has been merged.

**Pull request closed without merging**